File: RobHouse.py
from typing import List
class Solution:
    def rob(self, nums: List[int]) -> int:
        dp = [0 for i in range(len(nums)+1)]
        dp[1] = nums[0]
        for i in range(2,len(nums)+1):
            dp[i] = max(nums[i-1]+dp[i-2], dp[i-1])
        return max(dp[-1], dp[-2])

File: robhouse.java
public class robhouse {     
    public int rob(int[] nums) {         
        if(nums == null || nums.length ==0){             
            return 0;         
        }         
        int n = nums.length;         
        int skip=0;         
        int take = nums[0];          
        for(int i=1;i<n;i++){             
            int temp = skip;             
            skip = Math.max(skip,take);             
            take=temp+nums[i];         
        }         
        return Math.max(skip,take);     
    } 
}